index.js

```diff
@@ -25,9 +25,46 @@ db.connect();
 // express is used to connect .css and .js files
 app.use(express.static(path.join(__dirname, '/client/build')));
 
-// Label max is the number of labels a video needs to be considered fully labeled
+
+/***** Helping Functions and Variables *****/
+// LABEL_MAX is the number of labels a video needs to be considered fully labeled
 const LABEL_MAX = 5;
 
+// TIMEOUT is how long a video may be checked out in milliseconds (first number is minutes, 60000 converts to ms)
+const TIMEOUT = 10 * 60000
+
+// Create flatMap function to use when processing videos
+const myConcat = (x, y) =>
+  x.concat(y);
+const myFlatMap = (f, xs) => 
+  xs.map(f).reduce(myConcat, []);
+Array.prototype.myFlatMap = function(f) {
+  return myFlatMap(f, this);
+}
+
+// Checked out videos used to prevent videos from exceeding LABEL_MAX by simultaneous viewing
+checkedOutVideos = [];
```

Comment:
Im pretty sure this will need to be changed to `var checkedOutVideos  = [];` , I cant seem to find an initialize.